core/src/Revolution/modPhpThumb.php

```diff
@@ -332,13 +333,25 @@ public function SendSaveAsFileHeaderIfNeeded()
         if (headers_sent()) {
             return false;
         }
-        $downloadfilename = phpthumb_functions::SanitizeFilename(@$_GET['sia'] ? $_GET['sia'] : (@$_GET['down'] ? $_GET['down'] : 'phpThumb_generated_thumbnail' . (@$_GET['f'] ? $_GET['f'] : 'jpg')));
-        if (@$downloadfilename) {
-            $this->DebugMessage('SendSaveAsFileHeaderIfNeeded() sending header: Content-Disposition: ' . (@$_GET['down'] ? 'attachment' : 'inline') . '; filename="' . $downloadfilename . '"',
-                __FILE__, __LINE__);
-            header('Content-Disposition: ' . (@$_GET['down'] ? 'attachment' : 'inline') . '; filename="' . $downloadfilename . '"');
+    # Fix PHP warning: "Undefined array key sia"
+    if (isset($_GET['sia'])) {
+        $downloadfilename = phpthumb_functions::SanitizeFilename($_GET['sia']);
+    } elseif (isset($_GET['down'])) {
+        $downloadfilename = phpthumb_functions::SanitizeFilename($_GET['down']);
+    } else {
+        $downloadfilename = 'phpThumb_generated_thumbnail' . (isset($_GET['f']) ? $_GET['f'] : 'jpg');
+    } 
+   
+    # Fix PHP warning: "Undefined array key down"
+    if (isset($downloadfilename)) {
+        if (isset($_GET['down'])) {
+            $this->DebugMessage('SendSaveAsFileHeaderIfNeeded() sending header: Content-Disposition: attachment; filename="' . $downloadfilename . '"', __FILE__, __LINE__);
+            header('Content-Disposition: attachment; filename="' . $downloadfilename . '"');
+        } else {
+            $this->DebugMessage('SendSaveAsFileHeaderIfNeeded() sending header: Content-Disposition: inline; filename="' . $downloadfilename . '"', __FILE__, __LINE__);
+            header('Content-Disposition: inline; filename="' . $downloadfilename . '"');
         }
```

Comment:
Consider this alternative, which keeps the code more concise (like it originally was) but improves readability like you were going for:
```suggestion
        $inlineFilename = $_GET['sia'] ?? false;
        $attachmentFilename = $_GET['down'] ?? false;
        if ($inlineFilename || $attachmentFilename) {
            $filename = $inlineFilename ? $inlineFilename : $attachmentFilename ;
            $filename = phpthumb_functions::SanitizeFilename($filename);
        } else {
            $filename = 'phpThumb_generated_thumbnail' . ($_GET['f'] ?? 'jpg');
        }
        if ($filename) {
            $disposition = $attachmentFilename ? 'attachment' : 'inline' ;
            $this->DebugMessage('SendSaveAsFileHeaderIfNeeded() sending header: Content-Disposition: ' . $disposition . '; filename="' . $filename . '"', __FILE__, __LINE__);
            header('Content-Disposition: ' . $disposition . '; filename="' . $filename . '"');
        }
```